tests/virtctl/create_instancetype.go

```diff
@@ -0,0 +1,117 @@
+		DescribeTable("when CPU and Memory defined", func(namespaced string) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "2"),
+				setFlag(MemoryFlag, "256Mi"),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec, err := getInstancetypeSpec(namespaced, bytes, virtClient)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(instancetypeSpec.CPU.Guest).To(Equal(uint32(2)))
+			Expect(instancetypeSpec.Memory.Guest).To(Equal(resource.MustParse("256Mi")))
+		},
+			Entry("VirtualMachineInstancetype", namespaced),
+			Entry("VirtualMachineClusterInstancetype", ""),
+		)
+
+		DescribeTable("when GPUs defined", func(namespaced string) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "2"),
+				setFlag(MemoryFlag, "256Mi"),
+				setFlag(GPUFlag, "name:gpu1,devicename:nvidia/gpu1"),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec, err := getInstancetypeSpec(namespaced, bytes, virtClient)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(instancetypeSpec.GPUs[0].Name).To(Equal("gpu1"))
+			Expect(instancetypeSpec.GPUs[0].DeviceName).To(Equal("nvidia/gpu1"))
+		},
+			Entry("VirtualMachineInstancetype", namespaced),
+			Entry("VirtualMachineClusterInstancetype", ""),
+		)
+
+		DescribeTable("when hostDevice defined", func(namespaced string) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "2"),
+				setFlag(MemoryFlag, "256Mi"),
+				setFlag(HostDeviceFlag, "name:device1,devicename:hostdevice1"),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec, err := getInstancetypeSpec(namespaced, bytes, virtClient)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(instancetypeSpec.HostDevices[0].Name).To(Equal("device1"))
+			Expect(instancetypeSpec.HostDevices[0].DeviceName).To(Equal("hostdevice1"))
+		},
+			Entry("VirtualMachineInstancetype", namespaced),
+			Entry("VirtualMachineClusterInstancetype", ""),
+		)
+
+		DescribeTable("when IOThreadsPolicy defined", func(namespaced, policyStr string, policy v1.IOThreadsPolicy) {
+			bytes, err := clientcmd.NewRepeatableVirtctlCommandWithOut(create, Instancetype, namespaced,
+				setFlag(CPUFlag, "2"),
+				setFlag(MemoryFlag, "256Mi"),
+				setFlag(IOThreadsPolicyFlag, policyStr),
+			)()
+			Expect(err).ToNot(HaveOccurred())
+
+			instancetypeSpec, err := getInstancetypeSpec(namespaced, bytes, virtClient)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(*instancetypeSpec.IOThreadsPolicy).To(Equal(policy))
+		},
+			Entry("VirtualMachineInstancetype", namespaced, "auto", v1.IOThreadsPolicyAuto),
+			Entry("VirtualMachineClusterInstancetype", "", "shared", v1.IOThreadsPolicyShared),
+		)
+	})
+})
+
+func getInstancetypeSpec(namespaced string, bytes []byte, virtClient kubecli.KubevirtClient) (*instancetypev1alpha2.VirtualMachineInstancetypeSpec, error) {
```

Comment:
`createInstancetype` ?

Reply:
This name is already used, so maybe `createInstancetypeSpec` would be better than `getInstancetypeSpec`?